app/models/manageiq/providers/google/resource.rb

```diff
@@ -0,0 +1,40 @@
+module ManageIQ::Providers::Google
+  class Resource
+    def initialize(uri)
+      @type = self.class.parse_resource_type(uri)
+    end
+
+    attr_reader :type
```

Comment:
How many collection types are there? And how often will they change? I would prefer being as explicit as possible.
The whole idea of not passing strings around is that you have one central gatekeeper to check on the resource URIs.